yt/fields/tests/test_field_access.py

```diff
@@ -35,3 +39,44 @@ def test_field_access():
             plot_object._setup_plots()
             if hasattr(plot_object, "_frb"):
                 plot_object._frb[field]
+
+
+def test_unexisting_field_access():
+    ds = fake_random_ds(16, particles=10)
+
+    fname2ftype = defaultdict(list)
+
+    for ft, fn in ds.derived_field_list:
+        fname2ftype[fn].append(ft)
+
+    ad = ds.all_data()
+
+    ftypes = ("gas", "io")
+    fnames = (
+        "density",
+        "particle_position_x",
+        "particle_position_y",
+        "particle_position_z",
+    )
+
+    # Try invalid ftypes, fnames combinations
+    for ft, fn in product(ftypes, fnames):
+        if (ft, fn) in ds.derived_field_list:
+            continue
+        with assert_raises(YTFieldNotFound) as ex:
+            ad[(ft, fn)]
+
+        # Make sure the existing field has been suggested
+        for possible_ft in fname2ftype[fn]:
+            assert (possible_ft, fn) in ex.exception.suggestions
```

Comment:
I don't think we should test an implementation detail here (using a unique `suggestions` attribute), but rather check the actual error message. I believe you can do this in pytest using `pytest.raises` instead of `assert_raises`.
The test would need to be moved to a new file to ease transition towards pytest though.

Reply:
Leaving that for now ; I have to come up with a neat way of not stepping on @jcoughlin11's feet.

Reply:
Actually I went ahead and did it. Hoping tests pass.